src/transport/transport.h

```diff
@@ -224,33 +231,37 @@ class ConvectionTransport : public TransportBase {
      */
     EqData data_;
 
+    //@{
     /**
-     * Indicates if we finished the matrix and add vector by scaling with timestep factor.
+     * Flag indicates the state of object (transport matrix or source term).
+     * If false, the object is freshly assembled and not rescaled.
+     * If true, the object is scaled (not necessarily with the current time step).
      */
-	bool is_convection_matrix_scaled, need_time_rescaling;
-
-    double *sources_corr;
-    Vec v_sources_corr;
+	bool is_convection_matrix_scaled, is_src_term_scaled;
+    //@}
+    
+    double **sources_corr;
+    Vec *v_sources_corr;
     
-    ///temporary arrays to store constant values of fields over time interval
-    //(avoiding calling "field.value()" too often)
-    double **sources_density, 
-           **sources_conc,
-           **sources_sigma;
 
     TimeMark::Type target_mark_type;    ///< TimeMark type for time marks denoting end of every time interval where transport matrix remains constant.
-    double cfl_max_step;
-            // only local part
-
+    double cfl_max_step;    ///< Time step constraint coming from CFL condition.
+    
+    Vec vcfl_flow_,     ///< Parallel vector for flow contribution to CFL condition.
```

Comment:
Can we create these CFL related vectors temporary in asses_time_constraint?
Why there is recalculation of cfl_flow_ in convection_matrix_mpi?